File: app/src/main/java/com/alwaystinkering/sandbot/model/web/SandBotInterface.java
package com.alwaystinkering.sandbot.model.web;

import okhttp3.RequestBody;
import okhttp3.ResponseBody;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Path;

public interface SandBotInterface {
    @GET("/exec/g28")
    Call<Result> home();

    @GET("/exec/pause")
    Call<Result> pause();

    @GET("/exec/resume")
    Call<Result> resume();

    @GET("/exec/stop")
    Call<Result> stop();

    @GET("/pattern/{name}")
    Call<Result> startPattern(@Path("name") String name);

    @GET("/sequence/{name}")
    Call<Result> startSequence(@Path("name") String name);

    @GET("/getsettings")
    Call<ResponseBody> getSettings();

    @GET("/status")
    Call<ResponseBody> getStatus();

    @GET("/filelist/")
    Call<FileListResult> listFiles();

    @GET("/files/{fsName}/{fileName}")
    Call<ParametricFile> getParametricFile(@Path("fsName") String fsName, @Path("fileName") String fileName);


    //@Headers({"Content-Type: application/json"})
    @POST("/postsettings")
    Call<ResponseBody> postSettings(@Body RequestBody json);

    @POST("/setled")
    Call<ResponseBody> setLed(@Body RequestBody json);

}

File: app/src/main/java/com/alwaystinkering/sandbot/data/SandBotInterface.kt
package com.alwaystinkering.sandbot.data

import okhttp3.RequestBody
import okhttp3.ResponseBody
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Path

import com.alwaystinkering.sandbot.utils.Annotations.Str
import com.alwaystinkering.sandbot.utils.Annotations.Json

interface SandBotInterface {

    //    @GET("/getsettings")
    //    Call<ResponseBody> getSettings();

    @GET("/status")
    fun getStatus(): Call<BotStatus>

    @GET("/exec/g28")
    fun home(): Call<CommandResult>

    @GET("/exec/pause")
    fun pause(): Call<CommandResult>

    @GET("/exec/resume")
    fun resume(): Call<CommandResult>

    @GET("/exec/stop")
    fun stop(): Call<CommandResult>

    @GET("/playFile/{fsName}/{fileName}")
    fun playFile(@Path("fsName") fsName: String, @Path("fileName") name: String): Call<CommandResult>

    @GET("/filelist/")
    fun listFiles(): Call<FileListResult>

    @GET("/files/{fsName}/{fileName}")
    fun getParametricFile(@Path("fsName") fsName: String, @Path("fileName") fileName: String): Call<ParametricFile>

    @Str
    @GET("/files/{fsName}/{fileName}")
    fun getThetaRhoFile(@Path("fsName") fsName: String, @Path("fileName") fileName: String): Call<String>


    //@Headers({"Content-Type: application/json"})
    //    @POST("/postsettings")
    //    Call<ResponseBody> postSettings(@Body RequestBody json);

    @POST("/setled")
    fun setLed(@Body json: RequestBody): Call<ResponseBody>

}

annotation class Str
